### PR TITLE
Fixed tests.

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/ConnectionsControllerTests.cs [hunk 1]
```diff
@@ -148,9 +147,9 @@ public async void GetConnections_ShouldOnlyReturnConnectionsInvolvingTheCompany(
         var response = await client.GetFromJsonAsync<ConnectionsResponse>("api/connections");
 
         response.Should().NotBeNull();
-        response.Result.Should().HaveCount(2);
-        response.Result.Any(x => x.CompanyTin == ownedConnection1.CompanyBTin).Should().BeTrue();
-        response.Result.Any(x => x.CompanyTin == ownedConnection2.CompanyATin).Should().BeTrue();
+        response?.Result.Should().HaveCount(2);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/Controllers/ConnectionsControllerTests.cs:150]: I think we should use `!` in tests

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs [hunk 2]
```diff
@@ -261,12 +261,12 @@ await factory.SeedTransferAgreements(new List<TransferAgreement>()
 
         getTransferAgreement.Should().NotBeNull();
 
-        getTransferAgreement.Id.Should().Be(fakeTransferAgreement.Id);
-        getTransferAgreement.ReceiverTin.Should().Be(fakeTransferAgreement.ReceiverTin);
-        getTransferAgreement.StartDate.Should().Be(fakeTransferAgreement.StartDate.ToUnixTimeSeconds());
-        getTransferAgreement.EndDate.Should().Be(fakeTransferAgreement.EndDate?.ToUnixTimeSeconds());
-        getTransferAgreement.SenderName.Should().Be(fakeTransferAgreement.SenderName);
-        getTransferAgreement.SenderTin.Should().Be(fakeTransferAgreement.SenderTin);
+        getTransferAgreement?.Id.Should().Be(fakeTransferAgreement.Id);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs:264]: !

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs [hunk 3]
```diff
@@ -352,7 +352,7 @@ public async Task GetBySubjectId_ShouldReturnTransferAgreements_WhenUserHasTrans
         var transferAgreementsResponse = JsonConvert.DeserializeObject<TransferAgreementsResponse>(transferAgreements);
 
         transferAgreementsResponse.Should().NotBeNull();
-        transferAgreementsResponse.Result.Should().HaveCount(2);
+        transferAgreementsResponse?.Result.Should().HaveCount(2);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs:355]: !

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs [hunk 4]
```diff
@@ -527,7 +527,7 @@ public async Task EditEndDate_ShouldUpdateTransferAgreement_WhenInputIsValid()
 
         var updatedTransferAgreement = await response.Content.ReadFromJsonAsync<TransferAgreementDto>();
         updatedTransferAgreement.Should().NotBeNull();
-        updatedTransferAgreement.EndDate.Should().Be(newEndDate);
+        updatedTransferAgreement?.EndDate.Should().Be(newEndDate);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs:530]: !

File: domains/transfer/Transfer.API/API.IntegrationTests/Services/ConnectionInvitationCleanupServiceTests.cs [hunk 5]
```diff
@@ -61,6 +60,6 @@ public async Task Run_ShouldDeleteOldInvitations_WhenInvoked()
 
         var invitations = await dbContext.RepeatedlyQueryUntilCountIsMet<ConnectionInvitation>(1);
 
-        invitations.FirstOrDefault().Id.Should().Be(newInvitation.Id);
+        invitations.FirstOrDefault()?.Id.Should().Be(newInvitation.Id);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/Services/ConnectionInvitationCleanupServiceTests.cs:63]: !

File: domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs [hunk 6]
```diff
@@ -42,7 +42,7 @@ public async Task Run_WhenFail_CacheIsSetToUnhealthy()
         var service = new TransferAgreementsAutomationService(logger, transferAgreementRepository,
             projectOriginWalletService, memoryCache, metrics);
 
-        service.Run(cts.Token).Should().Throws(new TaskCanceledException());
+        await Assert.ThrowsAsync<TaskCanceledException>(() => service.Run(cts.Token));
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs:45]: Please use fluent assertion

File: domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs [hunk 7]
```diff
@@ -76,7 +76,7 @@ public async Task Run_WhenCalled_CacheIsSetToHealthy()
...
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs:79]: Fluent Assertion

File: domains/transfer/Transfer.API/API/ApiModels/Requests/CreateTransferAgreementValidator.cs [hunk 8]
```diff
@@ -31,7 +31,7 @@ public CreateTransferAgreementValidator(IHttpContextAccessor context)
             .Length(8)
             .Matches("^[0-9]{8}$")
             .WithMessage("ReceiverTin must be 8 digits without any spaces.")
-            .NotEqual(context.HttpContext.User.FindSubjectTinClaim())
+            .NotEqual(context.HttpContext?.User.FindSubjectTinClaim())
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/ApiModels/Requests/CreateTransferAgreementValidator.cs:34]: !

File: domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs (hunk 7 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/abafb4e698020d9fa926aea8a77832be16ab63ac

File: domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs (hunk 6 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.UnitTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/abafb4e698020d9fa926aea8a77832be16ab63ac

File: domains/transfer/Transfer.API/API/ApiModels/Requests/CreateTransferAgreementValidator.cs (hunk 8 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/ApiModels/Requests/CreateTransferAgreementValidator.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/29e0783358ba42e2edaad38de3d4221eebdc2d22

File: domains/transfer/Transfer.API/API.IntegrationTests/Services/ConnectionInvitationCleanupServiceTests.cs (hunk 5 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Services/ConnectionInvitationCleanupServiceTests.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/29e0783358ba42e2edaad38de3d4221eebdc2d22

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs (hunk 4 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/29e0783358ba42e2edaad38de3d4221eebdc2d22

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs (hunk 3 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/29e0783358ba42e2edaad38de3d4221eebdc2d22

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs (hunk 2 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/29e0783358ba42e2edaad38de3d4221eebdc2d22

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/ConnectionsControllerTests.cs (hunk 1 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Controllers/ConnectionsControllerTests.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/462/commits/29e0783358ba42e2edaad38de3d4221eebdc2d22

**Review by ckr123 — APPROVED** [review]

**Pull request merged**